src/components/v2v3/V2V3Project/V2V3PayButton/V2V3ConfirmPayModal/V2V3ConfirmPayModal.tsx

```diff
@@ -135,10 +140,18 @@ export function V2V3ConfirmPayModal({
     } = form.getFieldsValue()
 
     const txBeneficiary = beneficiary ?? userAddress
-    const delegateMetadata = encodeJB721DelegatePayMetadata({
-      ...(payProjectForm?.payMetadata as JB721DelegatePayMetadata),
-      ...payMetadataOverrides(projectId),
-    })
+
+    const delegateMetadata =
+      nftContractVersion === '1' // old delegate v1
+        ? encodeJB721DelegateV1PayMetadata({
+            ...(payProjectForm?.payMetadata as JB721DelegateV1PayMetadata),
+            ...payMetadataOverrides(projectId),
+          })
+        : // new delegate v1.1
+          encodeJB721DelegateV1_1PayMetadata({
```

Comment:
**issue** i think we should specifically only do this if      `nftContractVersion === '1.1'`. Might save our ass in the future.

Reply:
Good call, done